backend/src/db/cloud_storage.py

```diff
@@ -158,3 +158,20 @@ def retrieve_file(self, bucket_name: str, filename: str) -> bytes:
                 status_code=500,
                 detail=f"Failed to retrieve file '{filename}' from bucket '{bucket_name}': {e}",
             )
+
+    def delete_file(self, bucket_name: str, filename: str):
+        if not self.client.bucket_exists(bucket_name):
+            raise HTTPException(
+                status_code=404, detail=f"Bucket '{bucket_name}' does not exist."
+            )
+        try:
+            self.client.remove_object(bucket_name, filename)
+        except S3Error as e:
+            logger.error(
+                "Error deleting file '%s' from bucket '%s': %s"
+                % (filename, bucket_name, e)
+            )
+            raise HTTPException(
+                status_code=500,
+                detail=f"Failed to delete file '{filename}' from bucket '{bucket_name}': {e}",
+            )
```

Comment:
This task need to be continued here 
https://github.com/bh-coders/trip-planer/issues/108

Reply:
i need to think about this, how make global method with thread and delay to it.

Reply:
i have idea and this works but this is a bad way
`def publish_handler_event(
    event_type: str,
    data: dict,
    cache_handler: ICacheHandler,
    delay: str,
) -> None:
    try:
        logger.info("Start cache storage for publish")
        if delay:
            delay_time(delay)
        cache_handler.publish_event(
            event_type=event_type,
            data=data,
        )
        logger.info(
            "Cache storage for published to event_type %s data %s", event_type, data
        )
    except Exception as e:
        logger.error("Failed to handle publish event: %s", e)


def publish_handler_thread(
    event_type: str,
    data: dict,
    cache_handler: ICacheHandler,
    delay: str = None,
):
    try:
        thread = threading.Thread(
            target=publish_handler_event,
            args=(event_type, data, cache_handler, delay),
        )
        thread.start()
        logger.info("Handler thread %s started", thread.name)
        return True

    except Exception as e:
        logger.error("Failed to publish handler thread: %s", e)
        return False`

Reply:
>??